rust/example/src/main.rs

```diff
@@ -71,14 +76,53 @@ impl Counter for CounterImpl {
     }
 }
 
+
 #[derive(Debug, Parser)]
 struct Opt {
     #[clap(short, long, default_value = "eth0")]
     iface: String,
 }
 
+
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
+    if DEBUG {
```

Comment:
What is the reason for moving this behind DEBUG? Is it because the ticket was to get an example working and you did not want to break the frontend app?
If so, it is fine, however after this is merged, we definetly have to have a meeting together with @BenediktZinn, @ffranzgitHub and @Mr-Kanister to port stuff to the new folder and remove legacy stuff. Do cleanup in general.

Reply:
yes, I primarily wanted to test the Probe for myself. And the loader function currently gets implemented in the new folder structure in a different file, that's why I didn't prioritize the code quality here and just used the DEBUG variable. This code will be deprecated most likely.
A cleanup is overdue, lets make up an appointment in discord.